docs/workspace/advanced_tasks.md

```diff
@@ -129,6 +128,216 @@ This will add the following line to [manifest file](../reference/pixi_manifest.m
 bar = { cmd = "python bar.py", cwd = "scripts" }
 ```
 
+## Task Arguments
+
+Tasks can accept arguments that can be referenced in the command. This provides more flexibility and reusability for your tasks.
+
+### Why Use Task Arguments?
+
+Task arguments make your tasks more versatile and maintainable:
+
+- **Reusability**: Create generic tasks that can work with different inputs rather than duplicating tasks for each specific case
+- **Flexibility**: Change behavior at runtime without modifying your pixi.toml file
+- **Clarity**: Make your task intentions clear by explicitly defining what values can be customized
+- **Validation**: Define required arguments to ensure tasks are called correctly
+- **Default values**: Set sensible defaults while allowing overrides when needed
+
+For example, instead of creating separate build tasks for development and production modes, you can create a single parameterized task that handles both cases.
+
+Arguments can be:
+
+- **Required**: must be provided when running the task
+- **Optional**: can have default values that are used when not explicitly provided
+
+### Defining Task Arguments
+
+Define arguments in your task using the `args` field:
+
+```toml title="pixi.toml"
+[tasks]
+# Task with required arguments
+greet = {
+    cmd = "echo Hello, {{ name }}!", # name must be provided by the user
+    args = ["name"]
+}
+
+# Task with optional arguments (default values)
+build = {
+    cmd = "echo Building {{ project }} in {{ mode }} mode",
+    args = [
+        # If not provided, "my-app" will be used
+        { "arg": "project", "default": "my-app" },
+        # If not provided, "development" will be used
+        { "arg": "mode", "default": "development" }
+    ]
+}
+
+# Task with mixed required and optional arguments
+deploy = {
+    cmd = "echo Deploying {{ service }} to {{ environment }}",
+    args = [
+        "service",                                    # Required argument, no default
+        { "arg": "environment", "default": "staging" } # Optional with default value
+    ]
+}
```

Comment:
This is actually not valid TOML, since multi-line inline tables are not (yet) a thing. Please move that TOML to a file in `docs/source_files/pixi_tomls` and embed the relevant part as a snippet like this: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#embedding-external-files

We have tests, which ensure that these TOMLs can be parsed by pixi